### ISSUE TITLE
CONN_NOTICES_LIMIT should be configurable

### ISSUE THREAD

**Issue by dpage** [psycopg/psycopg2#464]
In pgAdmin 4 it's essential that we capture all notices from the server, as users rely on them for debugging and testing. Currently CONN_NOTICES_LIMIT is hard coded to 50. It would be good to make that a per-connection configurable, with a value of 0 meaning "do not trim the list, we'll do it from the app".

**Comment by dvarrazzo** [comment]
It makes sense.

**Comment by dvarrazzo** [comment]
Using a list for the notices and the notifications was a bad choice from the beginning: a queue would have been more efficient: for this reason I've made these attributes writable in psycopg 2.7. (see #326). Any object with an `append` method is supported. If the object is not a list or its subclass the 50 limit is not enforced.

For this reason, as soon as psycopg 2.7 is released, pgAdmin 4 may use e.g. a [deque](https://docs.python.org/2/library/collections.html#collections.deque) to store the notices, or even something more refined, such as an object exporting `append` as a callback to push the notice directly in a structure under pgAdmin4 control.

```python
In [24]: from collections import deque

In [25]: import psycopg2

In [26]: cnn = psycopg2.connect("")

In [27]: cnn.notices = deque()

In [28]: cur = cnn.cursor()

In [29]: for i in range(100):
   ....:     cur.execute("""do $$ begin raise notice 'hi'; end$$""")

In [30]: len(cnn.notices)
Out[30]: 100
```

I think the writable `notices` solves the problem. Because the limit would have been released in 2.7 anyway I think this issue can be closed.
